### PR TITLE
Added functions for posting, replying to posts, pinning a post, and marking a post resolved.

### PR DESCRIPTION
**Pull request by egregori3**
Added functions to support intelligent agents.

### CODE REVIEW

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 5fef1e10cf4ef238de6e3cc791e773986302789f
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: d9a87a3e5a4daefca2c3a21f6766fccb001950bd
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/hfaran/piazza-api

---
This message was auto-generated by https://gitcop.com

**Comment by egregori3** [comment]
I am not sure what I need to do to get this fork pulled. Any suggestions?

File: piazza_api/network.py [hunk 1]
```diff
@@ -195,6 +265,37 @@ def mark_as_duplicate(self, duplicated_cid, master_cid, msg=''):
...
+            cid = post
+
+        params = {
+            "cid": cid,
+            "resolved": "true"
+        }
+
+        return self._rpc.content_mark_resolved(params)
+
+    def pin(self, post):
```
Comment (hfaran) [inline piazza_api/network.py]: Can you document this using the same sphinx-style formatting as other methods?

Comment (hfaran) [reply piazza_api/network.py]: Also `resolve_post` might be a better method name

File: piazza_api/network.py [hunk 2]
```diff
@@ -195,6 +265,37 @@ def mark_as_duplicate(self, duplicated_cid, master_cid, msg=''):
         }
         return self._rpc.content_mark_duplicate(params)
 
+    def resolve(self, post):
```
Comment (hfaran) [inline piazza_api/network.py]: Please document this as well.

File: piazza_api/network.py [hunk 3]
```diff
@@ -173,6 +211,38 @@ def create_instructor_answer(self, post, content, revision, anonymous=False):
         }
         return self._rpc.content_instructor_answer(params)
 
+    def create_reply(self, post, content, anonymous=False):
+        """Create a reply to a followup
+        It seems like if the post has `<p>` tags, then it's treated as HTML,
+        but is treated as text otherwise. You'll want to provide `content`
+        accordingly.
```
Comment (hfaran) [inline piazza_api/network.py:220]: Add a newline below this.

**Review by hfaran — CHANGES_REQUESTED** [review]
Looks like you have a merge conflict - if you could address that and comments from this review, I will merge this. Thanks!

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 5fef1e10cf4ef238de6e3cc791e773986302789f
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: d9a87a3e5a4daefca2c3a21f6766fccb001950bd
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/hfaran/piazza-api

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 5fef1e10cf4ef238de6e3cc791e773986302789f
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: d9a87a3e5a4daefca2c3a21f6766fccb001950bd
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: 28fa79ed8cc97d7b424cbe3d687b0bb3b37d924f
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/hfaran/piazza-api

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 5fef1e10cf4ef238de6e3cc791e773986302789f
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: d9a87a3e5a4daefca2c3a21f6766fccb001950bd
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: 28fa79ed8cc97d7b424cbe3d687b0bb3b37d924f
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: 89120651d2362bfae0e412c000c1af2fc5bd5f30
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/hfaran/piazza-api

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 5fef1e10cf4ef238de6e3cc791e773986302789f
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: d9a87a3e5a4daefca2c3a21f6766fccb001950bd
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: 28fa79ed8cc97d7b424cbe3d687b0bb3b37d924f
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: 89120651d2362bfae0e412c000c1af2fc5bd5f30
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: d8556111c91e94b4df136e9042523efa950c7589
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/hfaran/piazza-api

---
This message was auto-generated by https://gitcop.com

**Comment by egregori3** [comment]
Should have all issues resolved. I ran pylint. I think I got everything. Tested on sandbox; loks good.

**Comment by egregori3** [comment]
>>> from piazza_api import Piazza
>>> p = Piazza()
>>> p.user_login()
Email: 
Password: 
>>> n = p.network("")
>>> post = n.get_post(27)
>>> n.delete_post(post)
'OK'
>>> post=n.get_post(28)
>>> n.pin_post(post)
{'item': {'folders': ['hw2'], 'nr': 28, 'main_version': 2, 'request_instructor': 0, 'log': [{'t': '2018-10-15T17:45:43Z', 'u': '', 'n': 'create'}, {'t': '2018-12-29T02:31:24Z', 'u': 'j', 'n': 'update'}], 'subject': 'adada', 'bucket_order': 0, 'no_answer_followup': 0, 'bucket_name': 'Pinned', 'num_favorites': 0, 'type': 'note', 'tags': ['hw2', 'instructor-note', 'pin'], 'log_p': [{'t': '2018-10-15T17:45:43Z', 'u': 'j', 'n': 'create'}, {'t': '2018-12-29T02:31:24Z', 'u': '', 'n': 'update'}], 'uid': '', 'unique_views': 6, 'content_snipet': 'abc\n#pin', 'view_adjust': 0, 'modified': '2018-12-29T02:31:24Z', 'id': '', 'updated': '2018-10-15T17:45:43Z', 'status': 'active'}}

>>> post=n.get_post(684)
>>> n.resolve_post(post['children'][0])
{'anon': 'no', 'folders': [], 'data': None, 'no_upvotes': 0, 'subject': '<p>azsSADFsadf</p>', 'created': '2018-12-29T02:32:43Z', 'bucket_order': 2, 'bucket_name': 'Today', 'type': 'followup', 'uid': '', 'children': [], 'no_answer': 0, 'id': '', 'updated': '2018-12-29T02:32:43Z', 'config': {}}

**Review by hfaran — APPROVED** [review]

**Comment by hfaran** [comment]
LGTM!

**Pull request merged**